plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -477,10 +480,22 @@ class producer_plugin_impl : public std::enable_shared_from_this<producer_plugin
             if( future.valid() ) {
                future.wait();
                app().post( priority::low, [self, future{std::move(future)}, persist_until_expired, next{std::move( next )}, trx_id]() mutable {
-                  auto exception_handler = [&next, trx_id](fc::exception_ptr ex) {
+                  auto exception_handler = [self, &next, trx_id](fc::exception_ptr ex) {
                     fc_dlog(_trx_failed_trace_log, "[TRX_TRACE] Speculative execution is REJECTING tx: ${txid} : ${why} ",
                             ("txid", trx_id)("why",ex->what()));
                      next(ex);
+
+                    if (_trx_trace_dump_log.is_enabled(fc::log_level::debug)) {
+                        if (self->_unapplied_transactions.get_trx(trx_id)) {
```

Comment:
Many/most of the time the transaction will not be in the unapplied_transactions queue. I think you are better off capturing a copy of `trx`. You could avoid capturing `trx_id` and just get that from `trx`.

Reply:
Thanks! Is it possible that function `get_trx(trx_id)` could return  a `nullptr`, i.e., trx_id is valid but there is no corresponding trx in the queue?

Reply:
yes

Reply:
Fixed.